source/common/common/scope_tracked_object_stack.h

```diff
@@ -21,23 +21,10 @@ class ScopeTrackedObjectStack : public ScopeTrackedObject {
   ScopeTrackedObjectStack(const ScopeTrackedObjectStack&) = delete;
   ScopeTrackedObjectStack& operator=(const ScopeTrackedObjectStack&) = delete;
 
-  void add(const ScopeTrackedObject& object) { tracked_objects_.push_back(object); }
-
-  OptRef<const StreamInfo::StreamInfo> trackedStream() const override {
-    for (auto iter = tracked_objects_.rbegin(); iter != tracked_objects_.rend(); ++iter) {
-      OptRef<const StreamInfo::StreamInfo> stream = iter->get().trackedStream();
-      if (stream.has_value()) {
-        return stream;
-      }
-    }
-    return {};
-  }
-
-  void dumpState(std::ostream& os, int indent_level) const override {
-    for (auto iter = tracked_objects_.rbegin(); iter != tracked_objects_.rend(); ++iter) {
-      iter->get().dumpState(os, indent_level);
-    }
-  }
+  void add(const ScopeTrackedObject& object);
```

Comment:
Note we're taking away inlining from these functions by doing this; considering how often this is used we should still inline them

Reply:
yeah on second thought i think i would rather refactor around this in the first place ... thanks for the comment 

/wait